graphdatascience/graph/graph_object.py

```diff
@@ -230,3 +231,194 @@ def __repr__(self) -> str:
+        if size_property is None:
+            size_property = str(uuid4())
+            self._query_runner.call_procedure(
+                endpoint="gds.pageRank.mutate",
+                params=CallParameters(graph_name=visual_graph, config=dict(mutateProperty=size_property)),
+                custom_error=False,
+            )
+            clean_up_size_prop = True
+        else:
+            clean_up_size_prop = False
+
+        node_properties = [size_property]
+        if include_node_properties is not None:
+            node_properties.extend(include_node_properties)
+
+        if color_property is not None:
+            node_properties.append(color_property)
+
+        # Remove possible duplicates
+        node_properties = list(set(node_properties))
+
+        result = self._query_runner.call_procedure(
+            endpoint="gds.graph.nodeProperties.stream",
+            params=CallParameters(
+                graph_name=visual_graph,
+                properties=node_properties,
+                nodeLabels=self.node_labels(),
+                config=dict(listNodeLabels=True),
+            ),
+            custom_error=False,
+        )
+
+        # new format was requested, but the query was run via Cypher
+        if "propertyValue" in result.keys():
+            wide_result = result.pivot(index=["nodeId"], columns=["nodeProperty"], values="propertyValue")
+            # nodeLabels cannot be an index column of the pivot as its not hashable
+            # so we need to manually join it back in
+            labels_df = result[["nodeId", "nodeLabels"]].set_index("nodeId")
+            wide_result = wide_result.join(labels_df, on="nodeId")
+            result = wide_result.reset_index()
+            result.columns.name = None
+        node_properties_df = result
+
+        if rel_weight_property is None:
+            relationships_df = self._query_runner.call_procedure(
+                endpoint="gds.graph.relationships.stream",
+                params=CallParameters(graph_name=visual_graph),
+                custom_error=False,
+            )
+        else:
+            relationships_df = self._query_runner.call_procedure(
+                endpoint="gds.graph.relationshipProperty.stream",
+                params=CallParameters(graph_name=visual_graph, properties=rel_weight_property),
+                custom_error=False,
+            )
+
+        # Clean up
+        if visual_graph != self._name:
+            self._query_runner.call_procedure(
+                endpoint="gds.graph.drop",
+                params=CallParameters(graph_name=visual_graph),
+                custom_error=False,
+            )
+        elif clean_up_size_prop:
+            self._query_runner.call_procedure(
+                endpoint="gds.graph.nodeProperties.drop",
+                params=CallParameters(graph_name=visual_graph, nodeProperties=size_property),
+                custom_error=False,
+            )
```

Comment:
I would suggest to have its own `Graph.sample` method and warn if the graph is too big.
Would avoid side-effect, the user is not aware off.